include/wx/qt/defs.h

```diff
@@ -12,6 +12,19 @@
 
 typedef class QWidget *WXWidget;
 
+/* ------------------------------------------------------------------------- */
+/*  macros to return the real Qt widget type corresponding to wx widget type */
+/* ------------------------------------------------------------------------- */
+
+#define wxDECLARE_GET_QT_HANDLE(qtwidget)     			\
```

Comment:
It seems a pity to add yet another widely used macro, IMHO it really doesn't have enough advantages compared to just writing out the accessor in full to use it.

Also, this might be over-engineering, but it should be possible to use CRTP here instead, i.e.

```cpp
template <typename QTWIDGET, typename WXCLASS>
class wxQtHandleAccessor
{
public:
    using QtWidget = QTWIDGET;
    QtWidget* GetQtHandle() const { return static_cast<QtWidget*>(static_cast<WXCLASS*>(this)->QtGetWindow()); }
};
```

and then derive `wxFoo` from `wxQtHandleAccessor<QtFoo, wxFoo>` to get the implementation of this function.

Reply:
> Also, this might be over-engineering, but it should be possible to use CRTP here instead, i.e.
> 
> ```c++
> template <typename QTWIDGET, typename WXCLASS>
> class wxQtHandleAccessor
> {
> public:
>     using QtWidget = QTWIDGET;
>     QtWidget* GetQtHandle() const { return static_cast<QtWidget*>(static_cast<WXCLASS*>(this)->QtGetWindow()); }
> };
> ```
> 
> and then derive `wxFoo` from `wxQtHandleAccessor<QtFoo, wxFoo>` to get the implementation of this function.

I did something similar before making this PR but it requires the corresponding Qt header to be included for the cast to succeed.

I just force pushed my initial version of the changes, but if you still have any suggestion to improve it, I'll happily implement it. TIA!